test/jasmine/tests/shapes_test.js

```diff
@@ -282,3 +285,518 @@ describe('Test shapes: a plot with shapes and an overlaid axis', function() {
         Plotly.plot(gd, data, layout).then(done);
     });
 });
+
+describe('Test shapes', function() {
+    'use strict';
+
+    var gd, data, layout, config;
+
+    beforeEach(function() {
+        gd = createGraphDiv();
+        data = [{}];
+        layout = {};
+        config = {
+            editable: true,
+            displayModeBar: false
+        };
+    });
+
+    afterEach(destroyGraphDiv);
+
+    var testCases = [
+        // xref: 'paper', yref: 'paper'
+        {
+            title: 'linked to paper'
+        },
+
+        // xaxis.type: 'linear', yaxis.type: 'log'
+        {
+            title: 'linked to linear and log axes',
+            xaxis: { type: 'linear', range: [0, 10] },
+            yaxis: { type: 'log', range: [Math.log10(1), Math.log10(1000)] }
+        },
+
+        // xaxis.type: 'date', yaxis.type: 'category'
+        {
+            title: 'linked to date and category axes',
+            xaxis: { type: 'date', range: ['2000-01-01', '2000-01-02'] },
```

Comment:
Hmm. `range` for date axes currently only work when set to timestamps. See example: http://codepen.io/etpinard/pen/RRoQKr